src/Types/Rational.purs

```diff
@@ -1,11 +1,94 @@
 module Types.Rational
   ( Rational
+  , class RationalComponent
+  , reduce
+  , (%)
+  , recip
+  , numerator
+  , denominator
+  , denominatorAsNat
   ) where
 
+import Prelude
+
+import Data.BigInt (BigInt, fromInt) as BI
+import Data.Maybe (Maybe(Just, Nothing), fromJust)
 import Data.Ratio (Ratio)
-import Data.BigInt (BigInt)
+import Data.Ratio ((%), numerator, denominator) as R
```

Comment:
I'm not really a fan of single-letter qualified imports, especially when they differ between modules (e.g. `Types.Rational` is imported as `R` in another module)